src/sentry/api/endpoints/organization_events_trace.py

```diff
@@ -262,16 +255,24 @@ def serialize(
 
                 # For the light response, the parent will be unknown unless it is a direct descendent of the root
                 is_root_child = root_span is not None
-                trace_results.append(
-                    self.serialize_event(
-                        snuba_event,
-                        root["id"] if is_root_child else None,
-                        1 if is_root_child else None,
+                # We only know to add the root if its the direct parent
+                if is_root_child:
+                    trace_results.append(
+                        self.serialize_event(
+                            root,
+                            None,
+                            0,
+                        )
                     )
-                )
+            else:
+                is_root_child = None
 
-            # The current event should be the last item in the trace_results
-            current_event = trace_results[-1]
+            current_event = self.serialize_event(
+                snuba_event,
+                root["id"] if is_root_child else None,
+                1 if is_root_child else 0 if root and root["id"] == snuba_event["id"] else None,
```

Comment:
Can we refactor this out of this one-liner?

Reply:
Very Pythonic.